### PR TITLE
EF 6 configuration with aspen 5

### LINKED ISSUE #633: Entity Framework 6 configuration
I have [started working](https://github.com/damienpontifex/Docs/tree/ef6-article) on a new topic for configuring and utilising EF6 in ASP.NET 5 after aspnet/Home#1062 in collaboration with @pgrudzien12 after his article http://bleedingnedge.com/2015/11/01/entity-framework-6-with-asp-net-5/

It covers:
- Connection strings and DI
- Configuration in code of EF6

### PR DESCRIPTION
**Pull request by damienpontifex**
Resolves #633 in discussing steps to utilise EF6 in an aspen 5 application.

### CODE REVIEW

**Comment by danroth27** [comment]
@rowanmiller Please review

File: aspnet/data/entity-framework-6.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+In this article:
+    - `Setup connection strings and dependency injection`_
+    - `Migrate configuration from config to code`_
+    - `Notes on Automatic Migrations`_
+    
+Prerequisites
+-------------
+    
+Before you start, make sure that you compile against full .NET Framework in your project.json as Entity Framework 6 does not support .NET Core. If you need cross platform features you will need to upgrade to Entity Framework 7.
```
Comment (rowanmiller) [inline aspnet/data/entity-framework-6.rst]: Can you include some info on how to do this, maybe even show the frameworks part of the project.json file.

File: aspnet/data/entity-framework-6.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+    
+SQL Server and SQL Server Express
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is the default and so no explicit configuration is needed. Optionally the above ``CodeConfig`` class can be used to explicitly set the connection factory and provider services.
+
+SQL Server Express LocalDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (rowanmiller) [inline aspnet/data/entity-framework-6.rst]: Given you are passing in a connection string, you shouldn't really need this. This is only needed if you are passing a database name and want EF to use the connection factory to work out which server to use. I wouldn't push folks down this path, but rather just tell them to pass in the whole connection string (as you have shown so far in the article).

File: aspnet/data/entity-framework-6.rst [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        public CodeConfig()
+        {
+            SetDefaultConnectionFactory(new System.Data.Entity.Infrastructure.LocalDbConnectionFactory("mssqllocaldb"));
+            SetProviderServices("System.Data.SqlClient",
+                System.Data.Entity.SqlServer.SqlProviderServices.Instance);
+        }
+    }
+
+Notes on Automatic Migrations
+-----------------------------
```
Comment (rowanmiller) [inline aspnet/data/entity-framework-6.rst]: You should remove "Automatic" from here... the note provided is about migrations in general and not just automatic migrations.

**Comment by rowanmiller** [comment]
This also needs to be rebased on the current master as there are merge conflicts. Ideally it should also be squashed as this is one cohesive addition but has 7 commits in it.

**Comment by rowanmiller** [comment]
I think the feedback needs to be addressed before we merge this, but it's good content and we should definitely accept it.

**Comment by damienpontifex** [comment]
Thanks @rowanmiller 

Have made the suggested changes. Am happy to rebase + squash into one commit. Will do this once signed off on the content so it can be successfully merged afterwards.

**Comment by danroth27** [comment]
@rowanmiller Are you signed off at this point?

**Comment by rowanmiller** [comment]
Yep :shipit:

**Comment by danroth27** [comment]
@shirhatti Please rebase and squash and then merge this one in.

**Comment by damienpontifex** [comment]
@shirhatti may take care of this, but so I learn the whole rebase/squash process, @danroth27 is this what should be done from here:

1. rebase from master
    * `git fetch upstream`
    * `git rebase upstream/master`
2. squash commits from our first commit in this article 1502f7338aeb73e9f76612ebecbea4e3918e9813
    * `git rebase -i 1502f73`
3. push back to my forked repo & branch
    * `git push origin +ef6-article`

**Comment by shirhatti** [comment]
@damienpontifex Yes! I think you have the process down.

Merged in 7c3e0ea4fa175308ae08fd73ca0c42d31708adf9
Thanks for the PR

**Pull request closed without merging**